JitTraceEnum_ELBO tests in test_jit are failing with CUDA

These tests are failing with CUDA, and giving a cryptic error about `TensorIterator expected type CUDADoubleType but got CPUDoubleType[2] (check_type_conversions at /home/npradhan/workspace/pyro_dev/pytorch/aten/src/ATen/native/TensorIterator.cpp:426)`.  This seems to get triggered in `Dice.compute_expectation`.

@eb8680 also observed the same error for the VAE example, but it was getting triggered from `Independent.log_prob` (I suppose the underlying reason is the same).

e.g. using `pytorch-0.5.0` branch with PyTorch master

```bash
CUDA_TEST=1 PYRO_TENSOR_TYPE=torch.cuda.DoubleTensor pytest -vs tests/infer/test_jit.py
```

<Details><summary>error trace</summary>

```
E       RuntimeError: 
E       TensorIterator expected type CUDADoubleType but got CPUDoubleType[2] (check_type_conversions at /home/npradhan/workspace/pyro_dev/pytorch/aten/src/ATen/native/TensorIterator.cpp:426)
E       frame #0: at::TensorIterator::Builder::build() + 0x5df (0x7f05dcc0814f in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/lib/libcaffe2.so)
E       frame #1: at::TensorIterator::binary_op(at::Tensor&, at::Tensor const&, at::Tensor const&) + 0x1f4 (0x7f05dcc08684 in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/lib/libcaffe2.so)
E       frame #2: at::native::mul_out(at::Tensor&, at::Tensor const&, at::Tensor const&) + 0x456 (0x7f05dcac9766 in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/lib/libcaffe2.so)
E       frame #3: at::native::mul(at::Tensor const&, at::Tensor const&) + 0x43 (0x7f05dcad1b13 in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/lib/libcaffe2.so)
E       frame #4: at::TypeDefault::mul(at::Tensor const&, at::Tensor const&) const + 0x75 (0x7f05dcdcb595 in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/lib/libcaffe2.so)
E       frame #5: torch::autograd::VariableType::mul(at::Tensor const&, at::Tensor const&) const + 0x2f3 (0x7f05cbc41f43 in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
E       frame #6: <unknown function> + 0x45ae4b (0x7f05cbcd6e4b in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
E       frame #7: <unknown function> + 0x4b2729 (0x7f05cbd2e729 in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
E       frame #8: <unknown function> + 0x491bd4 (0x7f05cbd0dbd4 in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/lib/libtorch.so.1)
E       frame #9: <unknown function> + 0x431d80 (0x7f05df466d80 in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/_C.cpython-36m-x86_64-linux-gnu.so)
E       frame #10: <unknown function> + 0x425eda (0x7f05df45aeda in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/_C.cpython-36m-x86_64-linux-gnu.so)
E       frame #11: <unknown function> + 0x42604e (0x7f05df45b04e in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/_C.cpython-36m-x86_64-linux-gnu.so)
E       frame #12: <unknown function> + 0x186b5c (0x7f05df1bbb5c in /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/_C.cpython-36m-x86_64-linux-gnu.so)
E       frame #13: _PyCFunction_FastCallDict + 0x154 (0x557211f239e4 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #14: _PyObject_FastCallDict + 0x2bf (0x557211f23dff in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #15: _PyObject_Call_Prepend + 0x63 (0x557211f28a73 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #16: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #17: <unknown function> + 0x16b897 (0x557211f7f897 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #18: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #19: _PyEval_EvalFrameDefault + 0x1abb (0x557211fd710b in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #20: <unknown function> + 0x196206 (0x557211faa206 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #21: _PyFunction_FastCallDict + 0x3d7 (0x557211fab897 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #22: _PyObject_FastCallDict + 0x26f (0x557211f23daf in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #23: _PyObject_Call_Prepend + 0x63 (0x557211f28a73 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #24: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #25: _PyEval_EvalFrameDefault + 0x1abb (0x557211fd710b in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #26: <unknown function> + 0x196206 (0x557211faa206 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #27: _PyFunction_FastCallDict + 0x1bc (0x557211fab67c in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #28: _PyObject_FastCallDict + 0x26f (0x557211f23daf in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #29: _PyObject_Call_Prepend + 0x63 (0x557211f28a73 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #30: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #31: <unknown function> + 0x16b897 (0x557211f7f897 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #32: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #33: _PyEval_EvalFrameDefault + 0x1abb (0x557211fd710b in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #34: <unknown function> + 0x19662e (0x557211faa62e in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #35: _PyFunction_FastCallDict + 0x1bc (0x557211fab67c in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #36: _PyObject_FastCallDict + 0x26f (0x557211f23daf in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #37: _PyObject_Call_Prepend + 0x63 (0x557211f28a73 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #38: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #39: <unknown function> + 0x16b897 (0x557211f7f897 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #40: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #41: _PyEval_EvalFrameDefault + 0x1abb (0x557211fd710b in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #42: <unknown function> + 0x19662e (0x557211faa62e in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #43: _PyFunction_FastCallDict + 0x3d7 (0x557211fab897 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #44: _PyObject_FastCallDict + 0x26f (0x557211f23daf in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #45: _PyObject_Call_Prepend + 0x63 (0x557211f28a73 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #46: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #47: _PyEval_EvalFrameDefault + 0x1abb (0x557211fd710b in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #48: <unknown function> + 0x196206 (0x557211faa206 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #49: <unknown function> + 0x1971cf (0x557211fab1cf in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #50: <unknown function> + 0x19ced5 (0x557211fb0ed5 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #51: _PyEval_EvalFrameDefault + 0x2fa (0x557211fd594a in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #52: PyEval_EvalCodeEx + 0x329 (0x557211fabcb9 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #53: <unknown function> + 0x198bd6 (0x557211facbd6 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #54: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #55: _PyEval_EvalFrameDefault + 0x1abb (0x557211fd710b in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #56: PyEval_EvalCodeEx + 0x329 (0x557211fabcb9 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #57: <unknown function> + 0x198ac4 (0x557211facac4 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #58: PyObject_Call + 0x3e (0x557211f237ee in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #59: _PyEval_EvalFrameDefault + 0x1abb (0x557211fd710b in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #60: <unknown function> + 0x19662e (0x557211faa62e in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #61: <unknown function> + 0x1971cf (0x557211fab1cf in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #62: <unknown function> + 0x19ced5 (0x557211fb0ed5 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       frame #63: _PyEval_EvalFrameDefault + 0x10c5 (0x557211fd6715 in /home/npradhan/miniconda3/envs/pytorch-master/bin/python)
E       :
E       operation failed in interpreter:
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/distributions/dirichlet.py(82): log_prob
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/poutine/trace_struct.py(151): compute_log_prob
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/infer/enum.py(53): get_importance_trace
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/infer/traceenum_elbo.py(233): _get_trace
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/infer/traceenum_elbo.py(280): _get_traces
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/infer/traceenum_elbo.py(423): differentiable_loss
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/poutine/messenger.py(27): _wraps
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/ops/jit.py(53): compiled
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/jit/__init__.py(515): trace
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/ops/jit.py(56): __call__
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/infer/traceenum_elbo.py(429): loss_and_grads
E       /home/npradhan/workspace/pyro_dev/pyro/pyro/infer/svi.py(96): step
E       /home/npradhan/workspace/pyro_dev/pyro/tests/infer/test_jit.py(407): test_dirichlet_bernoulli
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/python.py(196): pytest_pyfunc_call
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/callers.py(180): _multicall
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(61): <lambda>
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(67): _hookexec
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/hooks.py(258): __call__
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/python.py(1431): runtest
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/runner.py(111): pytest_runtest_call
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/callers.py(180): _multicall
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(61): <lambda>
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(67): _hookexec
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/hooks.py(258): __call__
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/runner.py(183): <lambda>
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/runner.py(201): __init__
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/runner.py(185): call_runtest_hook
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/runner.py(161): call_and_report
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/runner.py(81): runtestprotocol
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/runner.py(66): pytest_runtest_protocol
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/callers.py(180): _multicall
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(61): <lambda>
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(67): _hookexec
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/hooks.py(258): __call__
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/main.py(236): pytest_runtestloop
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/callers.py(180): _multicall
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(61): <lambda>
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(67): _hookexec
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/hooks.py(258): __call__
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/main.py(215): _main
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/main.py(178): wrap_session
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/main.py(208): pytest_cmdline_main
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/callers.py(180): _multicall
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(61): <lambda>
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/manager.py(67): _hookexec
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/pluggy/hooks.py(258): __call__
E       /home/npradhan/miniconda3/envs/pytorch-master/lib/python3.6/site-packages/_pytest/config/__init__.py(64): main
E       /home/npradhan/miniconda3/envs/pytorch-master/bin/pytest(11): <module>

../../../miniconda3/envs/pytorch-master/lib/python3.6/site-packages/torch/jit/__init__.py:1093: RuntimeError
```
</Details>

--- a/tests/infer/test_jit.py
+++ b/tests/infer/test_jit.py
@@ -1,7 +1,6 @@
 from __future__ import absolute_import, division, print_function
 
 import logging
-import os
 import warnings
 
 import pytest
@@ -28,8 +27,6 @@ def constant(*args, **kwargs):
 
 
 logger = logging.getLogger(__name__)
-pytestmark = pytest.mark.skipif('CUDA_TEST' in os.environ,
-                                reason='https://github.com/uber/pyro/issues/1419')
 
 
 def test_simple():

